linode/manager
# test: [M3-7183] - Account for Managed test acccounts in Linode Backup tests

## Description 📝
This allows the Cypress backup tests to more gracefully handle cases where the test account has Managed.

- By default, tests that are impacted by Managed will be skipped if the test account has Managed enabled
  - At the moment only the Linode Backup tests are affected
- If the `CY_TEST_FAIL_ON_MANAGED` environment variable is defined, the tests will fail instead. This is intended for CI environments where Managed should never be enabled on test accounts.

See this discussion for more context: https://github.com/linode/manager/pull/9699#pullrequestreview-1635928325

## Changes  🔄
- Fetch and cache test account settings when Cypress launches to detect whether or not Managed is enabled
- Adds utilities to skip Cypress tests
- Adds `CY_TEST_FAIL_ON_MANAGED` environment variable to control test behavior on Managed accounts
- Skip/fail Linode Backup tests when test account has Managed enabled
- Fixes one of the backup integration tests by mocking account settings

## How to test 🧪
- We can rely on the automated tests to ensure that tests work as expected when test accounts do not have Managed enabled

### Prerequisites
- Enable Managed on your test account through Cloud Manager
  - (Feel free to reach out on Slack if you have any questions about enabling or disabling Managed)
- Run `yarn && yarn build && yarn start:manager:ci`

### Verification steps 
- Run `yarn cy:run -s "cypress/e2e/core/linodes/backup-linode.spec.ts"` and confirm that the "can enable backups" test is skipped and the rest of the tests pass
- Run `CY_TEST_FAIL_ON_MANAGED=1 yarn cy:run -s "cypress/e2e/core/linodes/backup-linode.spec.ts"` and confirm that the "can enable backups" test fails and the rest of the tests pass

## As an Author I have considered 🤔

*Check all that apply*
